### PR TITLE
[ENH] Table.attributes field.

### PR DESCRIPTION
**Pull request by markotoplak**

### CODE REVIEW

File: Orange/data/io.py [hunk 1]
```diff
@@ -674,6 +677,9 @@ def write_file(cls, filename, data):
             writer = csv.writer(file, delimiter=cls.DELIMITERS[0])
             cls.write_headers(writer.writerow, data)
             cls.write_data(writer.writerow, data)
+        if isinstance(filename, str) and data.attributes:
+            with open(filename + '.apck', 'wb') as f:
+                pickle.dump(data.attributes, f, pickle.HIGHEST_PROTOCOL)
```
Comment (kernc) [inline Orange/data/io.py]: Could these two chunks be methods on `FileFormat` class? There might be use for them with other readers/writers?

File: Orange/data/io.py [hunk 2]
```diff
@@ -662,6 +662,9 @@ def read(self):
                                    '{}{}').format(pos,
                                                   ('-' + str(endpos)) if (endpos - pos) > 1 else '')
                         warnings.warn(warning)
+                    if isinstance(self.filename, str) and path.exists(self.filename + '.apck'):
+                        with open(self.filename + '.apck', 'rb') as f:
```
Comment (kernc) [inline Orange/data/io.py]: I vote for _.attributes_ or _.metadata_ file extensions.

File: Orange/data/table.py [hunk 3]
```diff
@@ -364,6 +369,10 @@ def from_table_rows(cls, source, row_indices):
         self.W = source.W[row_indices]
         self.name = getattr(source, 'name', '')
         self.ids = np.array(source.ids[row_indices])
+        if hasattr(source, "attributes"):
+            self.attributes = source.attributes
+        else:
+            self.attributes = {}
```
Comment (kernc) [inline Orange/data/table.py]:
```py
self.attributes = getattr(source, 'attributes', {})
```

File: Orange/data/table.py [hunk 4]
```diff
@@ -334,6 +335,10 @@ def get_columns(row_indices, src_cols, n_rows, dtype=np.float64):
                 self.ids = np.array(source.ids[row_indices])
             else:
                 cls._init_ids(self)
+            if hasattr(source, 'attributes'):
+                self.attributes = source.attributes
+            else:
+                self.attributes = {}
```
Comment (kernc) [inline Orange/data/table.py]: As below.

**Comment by BlazZupan** [comment]
I have tested this and works well. Please update the documentation as well (Data Table section of the Reference documentation).

**Comment by BlazZupan** [comment]
This feature is needed urgently. I have merged it, though documentation is still required. The request for documentation was in this case (as an exception) added to Orange to do list.

**Pull request merged**